src/main/scala/diplomacy/Parameters.scala

```diff
@@ -322,3 +322,23 @@ trait DirectedBuffers[T] {
   def copyOut(x: BufferParams): T
   def copyInOut(x: BufferParams): T
 }
+
+trait IdMapEntry {
+  val name: String
+  val from: IdRange
+  val to: IdRange
+  val isCache: Boolean
+  val requestFifo: Boolean
+  def pretty(fmt: String) =
+    if (from ne to) {
```

Comment:
Every implementer of `IdMapEntry` that I can see is a `case class` thus using structural equality. I see this is a "pretty" method so perhaps there's good reason to use referential equality but I would suggest documenting why that is because it is not immediately apparent to me.

Reply:
Because if a mapping coincidentally maps from the same values to the same values, it is still doing an identity transformation, whereas if they are literally an identical reference then no mapping was done. i will add a comment